pkg/crc/machine/kubeconfig_test.go

```diff
@@ -0,0 +1,39 @@
+    server: https://api.crc.testing:6443
+  name: crc
+contexts:
+- context:
+    cluster: crc
+    user: test
+  name: test
+current-context: test
+kind: Config
+preferences: {}
+users:
+- name: test
+  user: 
+    token: 85YUSkhhtVmXvB1ltw-Gx92sWhzjMm_5h6Na4A1IZK8
+`
+
+func TestGetClusterCACert(t *testing.T) {
+	f, err := ioutil.TempFile("", "kubeconfig")
+	assert.NoError(t, err, "")
+	_, err = f.WriteString(dummyKubeconfigFileContent)
+	assert.NoError(t, err, "")
+	st, err := GetClusterCACert(f.Name())
+	assert.NoError(t, err, "")
+	expectedString := "-----BEGIN CERTIFICATE-----\nMIIDODCCAiCgAwIBAgIIRVfCKNUa1wIwDQYJKoZIhvcNAQELBQAwJjEkMCIGA1UE\nAwwbaW5ncmVzcy1vcGVyYXRvckAxNTk5OTkxMDc4MB4XDTIwMDkxMzA5NTgwOFoX\nDTIyMDkxMzA5NTgwOVowHTEbMBkGA1UEAwwSKi5hcHBzLWNyYy50ZXN0aW5nMIIB\nIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAtClYnM1FYiH+wcYiGIUnLvNi\nzLWWjsqJrQ/4Snnu61HDaU5w3An9yDZojyaJdCmWUg6CKXDiCoJB+lxMFdyaolXU\ndohJ9vr2wt6iuNfshmtxUwiBhI9ZBsVhztWdu3cgnUcYW8KMyUmajiEyXD8Npvba\nZ4ifUsjAYE1LByZzPIkmBmKPnv0fFZu1ejgg7HuQlYmfN/pJHudWMvwZJ8b3Zhqn\nsA0cDsaCDU1SJk6cQvNJFgF38+IWNhJpiQlGfFwKkal64/RYcR9yv8/pMV+7jj1z\nBPI9yxbFgLuxLxlyS8Kxnz9ln/4V0ZB12qU3c6aM4Ew3uhYv8ZDSOnpTUSjrRwID\nAQABo3MwcTAOBgNVHQ8BAf8EBAMCBaAwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDAYD\nVR0TAQH/BAIwADAdBgNVHQ4EFgQUTtNCBUyuPpf4M4YhlYBqJhZGQYYwHQYDVR0R\nBBYwFIISKi5hcHBzLWNyYy50ZXN0aW5nMA0GCSqGSIb3DQEBCwUAA4IBAQBkGZiv\nBqqEDKUTiifFfTxQXJzO5OBBTUDqSntrknAw0sidPgn9A8a2gGdCr7mKEH16FQ7N\n1SpkjXWZghE/1pZTaS7JVcT6+Yuplfy5Reoim/Nlu8ulDDfBSSp9S9BO+Q/lFJdM\nomwyIu0sSXI79ColhLirDinEmyQMDHdmTJ+kJfctpSH27L4j5osJxtNSAEABmUgX\n1HQ7FYzrNmys5OcK9SlDn0vCyOTHyqBbxRnF6L9Ec2bU6KD0DvEBurBpAYGMc9ss\nBD0FXdHEPg7HP0Nep79jhe10IXGrghtep3D5jUjbj1I4DGSsQ8y4df2vo6IFd96A\nf0uUS/73sncN64gl\n-----END CERTIFICATE-----\n"
```

Comment:
Wondering if this should also be base64-encoded in the json we return with `-ojson`?

Reply:
@cfergeau if we make it base64 encoded than to have it in the file we again need to decode it and then have it which is one more step.

Reply:
in ~/.kube/config CA is base64, I guess we should follow them.

Reply:
@guillaumerose yes but if we parse that using client tool then the return []byte are decoded one, if you folks want the encoded one then we need to again encode it.

Reply:
I know base64 is inconvenient, but the certificates contains `\n` characters, I have no idea how well this is going to be supported if we put this unencoded in yaml data?

Reply:
Everything between the lines:  `-----BEGIN CERTIFICATE -----` and. `-----END CERTIFICATE -----` are already BASE64 encoded. 

I would instead here focus on the line ending issue... though, does YAML not support multi-line entries? Or what about just stripping the line endings?

Reply:
It is not encoded right now, if all agree I just encode it and then consumers need to decode it and use accordingly.